network/src/protocols/health_checker/test.rs

```diff
@@ -15,136 +15,148 @@ use crate::{
+        let mock_time = TimeService::mock();
+
+        let (peer_mgr_reqs_tx, peer_mgr_reqs_rx) = diem_channel::new(QueueStyle::FIFO, 1, None);
+        let (connection_reqs_tx, connection_reqs_rx) = diem_channel::new(QueueStyle::FIFO, 1, None);
+        let (peer_mgr_notifs_tx, peer_mgr_notifs_rx) = diem_channel::new(QueueStyle::FIFO, 1, None);
+        let (connection_notifs_tx, connection_notifs_rx) = conn_notifs_channel::new();
+
+        let hc_network_tx = HealthCheckerNetworkSender::new(
+            PeerManagerRequestSender::new(peer_mgr_reqs_tx),
+            ConnectionRequestSender::new(connection_reqs_tx),
+        );
+        let hc_network_rx =
+            HealthCheckerNetworkEvents::new(peer_mgr_notifs_rx, connection_notifs_rx);
+        let network_context =
+            NetworkContext::new(NetworkId::Validator, RoleType::Validator, PeerId::ZERO);
+        let health_checker = HealthChecker::new(
+            Arc::new(network_context),
+            mock_time.clone(),
+            hc_network_tx,
+            hc_network_rx,
+            PING_INTERVAL,
+            PING_TIMEOUT,
+            ping_failures_tolerated,
+        );
+
+        (
+            Self {
+                mock_time: mock_time.into_mock(),
+                peer_mgr_reqs_rx,
+                peer_mgr_notifs_tx,
+                connection_reqs_rx,
+                connection_notifs_tx,
+            },
+            health_checker,
+        )
+    }
 
-async fn expect_ping(
-    network_reqs_rx: &mut diem_channel::Receiver<(PeerId, ProtocolId), PeerManagerRequest>,
-) -> (Ping, oneshot::Sender<Result<Bytes, RpcError>>) {
-    let req = network_reqs_rx.next().await.unwrap();
-    let rpc_req = match req {
-        PeerManagerRequest::SendRpc(_peer_id, rpc_req) => rpc_req,
-        _ => panic!("Unexpected PeerManagerRequest: {:?}", req),
-    };
+    fn new_strict() -> (Self, HealthChecker) {
+        Self::new_permissive(0 /* ping_failures_tolerated */)
+    }
+
+    async fn trigger_ping(&self) {
+        self.mock_time.advance_async(PING_INTERVAL).await;
+    }
 
-    let protocol_id = rpc_req.protocol_id;
-    let req_data = rpc_req.data;
-    let res_tx = rpc_req.res_tx;
+    async fn expect_ping(&mut self) -> (Ping, oneshot::Sender<Result<Bytes, RpcError>>) {
+        let req = self.peer_mgr_reqs_rx.next().await.unwrap();
+        let rpc_req = match req {
+            PeerManagerRequest::SendRpc(_peer_id, rpc_req) => rpc_req,
+            _ => panic!("Unexpected PeerManagerRequest: {:?}", req),
+        };
 
-    assert_eq!(protocol_id, ProtocolId::HealthCheckerRpc,);
+        let protocol_id = rpc_req.protocol_id;
+        let req_data = rpc_req.data;
+        let res_tx = rpc_req.res_tx;
 
-    match bcs::from_bytes(&req_data).unwrap() {
-        HealthCheckerMsg::Ping(ping) => (ping, res_tx),
-        msg => panic!("Unexpected HealthCheckerMsg: {:?}", msg),
+        assert_eq!(protocol_id, ProtocolId::HealthCheckerRpc,);
```

Comment:
Nit: This always bothered me that the formatter, allows either basically hanging commas at the end, or no hanging commas, and didn't choose one or the other.  It also seemingly adds one randomly sometimes.

Reply:
ughh, yeah rustfmt chokes a bit when it hits a trailing comma in a macro. I think it's because custom macro rules may or may not require a trailing comma, so rustfmt can't necessarily remove it without potentially breaking something.